src/replay/freeze.zig

```diff
@@ -0,0 +1,563 @@
+        .{ params.hash_slot.slot, slot_hash.get().*.?.base58String().slice() },
+    );
+
+    // NOTE: agave updates hard_forks and hash_overrides here
+}
+
+const FinalizeStateParams = struct {
+    /// nested dependencies
+    update_sysvar: UpdateSysvarAccountDeps,
+
+    // shared state
+    account_store: AccountStore,
+    account_reader: SlotAccountReader,
+    capitalization: *std.atomic.Value(u64),
+    blockhash_queue: *sig.sync.RwMux(sig.core.BlockhashQueue),
+
+    // data params
+    rent: Rent,
+    slot: Slot,
+    blockhash: Hash,
+    lamports_per_signature: u64,
+    collector_id: Pubkey,
+    collected_transaction_fees: u64,
+    collected_priority_fees: u64,
+};
+
+/// Updates some accounts and other shared state to finish up the slot execution.
+fn finalizeState(allocator: Allocator, params: FinalizeStateParams) !void {
+    // Update recent blockhashes (NOTE: agave does this in registerTick)
+    {
+        var q = params.blockhash_queue.write();
+        defer q.unlock();
+        try q.mut().insertHash(allocator, params.blockhash, params.lamports_per_signature);
+    }
+    {
+        var q = params.blockhash_queue.read();
+        defer q.unlock();
+        try updateRecentBlockhashes(allocator, q.get(), params.update_sysvar);
+    }
+
+    try distributeTransactionFees(
+        allocator,
+        params.account_store,
+        params.account_reader,
+        params.capitalization,
+        params.rent,
+        params.slot,
+        params.collector_id,
+        params.collected_transaction_fees,
+        params.collected_priority_fees,
+    );
+
+    // Run incinerator
+    if (try params.account_reader.get(sig.runtime.ids.INCINERATOR)) |incinerator_account| {
+        _ = params.capitalization.fetchSub(incinerator_account.lamports, .monotonic);
+        try params.account_store.put(
+            params.update_sysvar.slot,
+            sig.runtime.ids.INCINERATOR,
+            .EMPTY,
+        );
+    }
+
+    try updateSlotHistory(allocator, params.update_sysvar);
+}
+
+/// Burn and payout the appropriate portions of collected fees.
+fn distributeTransactionFees(
+    allocator: Allocator,
+    account_store: AccountStore,
+    account_reader: SlotAccountReader,
+    capitalization: *std.atomic.Value(u64),
+    rent: Rent,
+    slot: Slot,
+    collector_id: Pubkey,
+    collected_transaction_fees: u64,
+    collected_priority_fees: u64,
+) !void {
+    var burn = collected_transaction_fees * 50 / 100;
+    const total_fees = collected_priority_fees + collected_transaction_fees;
+    const payout = total_fees -| burn;
+    tryPayoutFees(
+        allocator,
+        account_store,
+        account_reader,
+        rent,
+        slot,
+        collector_id,
+        payout,
+    ) catch |e|
+        switch (e) {
+            error.InvalidAccountOwner,
+            error.LamportOverflow,
+            error.InvalidRentPayingAccount,
+            => burn = total_fees,
+            else => return e,
+        };
```

Comment:
Interesting choice of indentation, any particular reason for this, or just aesthetics?

Reply:
it was a gradual evolution of changes. first i kept the arguments all on one line and moved the switch. then i added more arguments and needed to break the function call up into multiple lines, and overlooked that the catch and switch were still separated

62c929fc79624c14826336c2c9012386924f2342